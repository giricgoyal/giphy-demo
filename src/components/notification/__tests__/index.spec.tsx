import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Notification, { Props } from '../index'

describe('<Notification />', () => {
    let container
    const onCloseHandler = jest.fn()

    const defaultProps: Partial<Props> = {
        show: true,
        onClose: onCloseHandler,
    }

    describe('When show is true', () => {
        beforeEach(() => {
            ;({ container } = render(
                <Notification {...defaultProps}>
                    <div>I am Thor! God of thunder.</div>
                </Notification>,
            ))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When show is false', () => {
        beforeEach(() => {
            const props: Partial<Props> = {
                show: false,
            }

            ;({ container } = render(
                <Notification {...defaultProps} {...props}>
                    <div>I am Thor! God of thunder.</div>
                </Notification>,
            ))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
