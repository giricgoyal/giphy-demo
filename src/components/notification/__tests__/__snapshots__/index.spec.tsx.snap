// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> When show is false should match snapshot 1`] = `<div />`;

exports[`<Notification /> When show is true should match snapshot 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="fade toast notification bg-danger show"
    role="alert"
  >
    <div
      class="toast-body"
    >
      <div>
        I am Thor! God of thunder.
      </div>
    </div>
  </div>
</div>
`;
